crates/db-node/src/check_proof_state.rs

```diff
@@ -0,0 +1,27 @@
+use nuts::{
+    nut01::PublicKey,
+    nut07::{ProofCheckState, ProofState},
+};
+use sqlx::PgConnection;
+
+pub async fn get_proofs_by_y(
+    conn: &mut PgConnection,
+    ys: impl Iterator<Item = Vec<u8>>,
+) -> Result<Vec<ProofCheckState>, sqlx::Error> {
+    let records = sqlx::query!(
+        r#"SELECT y, state FROM proof WHERE y = ANY($1);"#,
+        &ys.collect::<Vec<_>>()
```

Comment:
we need the states in the same order as the inputs `ys`, this query doesn't guranatee this. You could reorder later on in the grpc function, but it will be way better to do it at the database level.
Take a look at `get_by_blind_secrets` in [crates/db-node/src/blind_signature.rs](https://github.com/nutty-raccoon/paynet/blob/0b6c3c55d532044bc691980728b630f182ba3799/crates/db-node/src/blind_signature.rs#L46) .

You don't need to return the `y` values from, DB, you already have them in your input